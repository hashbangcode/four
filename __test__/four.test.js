const {
  getCanvasContextObject,
  setupGrid,
  drawGrid,
  initCanvas,
} = require('../four');

describe('four tests:', () => {
  let canvas;

  beforeEach(() => {
    canvas = document.createElement('canvas');
    canvas.setAttribute('id', 'four');
    document.getElementsByTagName('body')[0].append(canvas);
  });

  test('test set up grid', () => {
    const theGrid = setupGrid();
    expect(theGrid.length).toStrictEqual(4);
    expect(theGrid[0].length).toStrictEqual(4);
  });

  test('test drawing grid', () => {
    initCanvas('four');
    const ctx = getCanvasContextObject();
    drawGrid();
    const events = ctx.__getEvents();
    expect(events).toMatchSnapshot();
  });
});
