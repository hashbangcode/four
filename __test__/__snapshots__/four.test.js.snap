// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`four tests: test drawing grid 1`] = `
[
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 75,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 150,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 225,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 0,
      "y": 75,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 75,
      "y": 75,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 150,
      "y": 75,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 225,
      "y": 75,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 0,
      "y": 150,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 75,
      "y": 150,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 150,
      "y": 150,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 225,
      "y": 150,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 0,
      "y": 225,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 75,
      "y": 225,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 150,
      "y": 225,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 225,
      "y": 225,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 75,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 150,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 225,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 0,
      "y": 75,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 75,
      "y": 75,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 150,
      "y": 75,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 225,
      "y": 75,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 0,
      "y": 150,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 75,
      "y": 150,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 150,
      "y": 150,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 225,
      "y": 150,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 0,
      "y": 225,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 75,
      "y": 225,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 150,
      "y": 225,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
  {
    "props": {
      "height": 37.5,
      "width": 75,
      "x": 225,
      "y": 225,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeRect",
  },
]
`;
